api/agent/protocol/json.go

```diff
@@ -0,0 +1,114 @@
+package protocol
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+// This is sent into the function
+// All HTTP request headers should be set in env
+type JSONIO struct {
```

Comment:
this can be lower cased, since it's only used locally for encoding / decoding and the caller does not see this. as long as the fields are exported, JSON will behave correctly.